src/main/java/com/commercetools/sync/commons/utils/EnumValuesUpdateActionUtils.java

```diff
@@ -0,0 +1,434 @@
+
+public final class EnumValuesUpdateActionUtils {
+
+    /**
+     * Compares a list of old {@code oldEnumValues} with a list of new {@code newEnumValues} for a given definition.
+     * The method serves as an implementation for enum values syncing. The method takes in functions
+     * for building the required update actions (AddEnumValue, ChangeEnumValueOrder
+     * and 1-1 update actions on enum values (e.g. changeLabel) for the required resource.
+     *
+     * @param definitionName                  the definition name whose enum values are going to be synced.
+     * @param oldEnumValues                   the old list of plain enum values.
+     * @param newEnumValues                   the new list of plain enum values.
+     * @param removeEnumCallback              the function that is called in order to remove the new enum instance.
+     * @param matchingEnumCallback            the function that is called to get the update action resulting from
+     *                                        comparing the enum value fields one by one.
+     * @param addEnumCallback                 the function that is called in order to add the new enum instance.
+     * @param changeOrderEnumCallback         the function that is called to apply the change in the order.
+     * @param changeOrderWithKeysEnumCallback the function that is called to apply the change in the order with keys.
+     * @param <T>                             the enum type of the elements to change the order for.
+     * @param <U>                             the type of the resource in which the update actions will be applied on.
+     * @return a list of enum values update actions if the list of plain enum values is not identical.
+     *         Otherwise, if the plain enum values are identical, an empty list is returned.
+     */
+    @Nonnull
+    public static <T extends WithKey, U> List<UpdateAction<U>> buildActions(
+        @Nonnull final String definitionName,
+        @Nonnull final List<T> oldEnumValues,
+        @Nullable final List<T> newEnumValues,
+        @Nullable final BiFunction<String, List<String>, UpdateAction<U>> removeEnumCallback,
+        @Nullable final TriFunction<String, T, T, List<UpdateAction<U>>> matchingEnumCallback,
+        @Nullable final BiFunction<String, T, UpdateAction<U>> addEnumCallback,
+        @Nullable final BiFunction<String, List<T>, UpdateAction<U>> changeOrderEnumCallback,
+        @Nullable final BiFunction<String, List<String>, UpdateAction<U>> changeOrderWithKeysEnumCallback) {
+
+        if (newEnumValues != null && !newEnumValues.isEmpty()) {
+            return buildUpdateActions(definitionName,
+                    oldEnumValues,
+                    newEnumValues,
+                    removeEnumCallback,
+                    matchingEnumCallback,
+                    addEnumCallback,
+                    changeOrderEnumCallback,
+                    changeOrderWithKeysEnumCallback
```

Comment:
For type resource the `ChangeLocalizedEnumValueOrder::of` takes `List<String> keys` as parameter instead of value like `List<LocalizedEnumValue> values `. So the type resource is using `changeOrderWithKeysEnumCallback ` callback to create `changeLocalizedEnumValueOrder` build action. this is kind of confusing but I just find this way to pass additional callback function which just takes keys. Do you have a better idea for this case?